src/lib/utils/stl_util.h

```diff
@@ -218,6 +218,8 @@ class BufferStuffer {
          std::copy(buffer.begin(), buffer.end(), sink.begin());
       }
 
+      void append(uint8_t b) { append({&b, 1}); }
```

Comment:
I realized (after the fact) that (AIUI) `next_byte` is the existing approach here. I *really* don't like that :joy: so I propose this.

Reply:
A fairly common idiom - at least in the `pk_pad` code is

```
   while(stuffer.remaining_capacity() > some_len) {
      stuffer.append(some_fixed_byte);
   }
```

and I'm a little tempted to add a helper for it

`stuffer.pad_until(some_len, some_fixed_byte)`

 and I imagine you will have some opinion on this matter.

Reply:
> I realized (after the fact) that (AIUI) next_byte is the existing approach here. I really don't like that 😂 so I propose this.

Works for me. I guess my mind was fully in "spanify-everything" mode when designing this. Returning a byte reference into the underlying buffer is semantically closer to `std::span<>`. But yeah, I agree. `append(uint8_t)` makes much more sense.